### PR TITLE
Introduce suspenders:generators:single_redirect

### PR DESCRIPTION
**Pull request by mike-burns**
Create generator for `Rack::CanonicalHost`.

### CODE REVIEW

File: spec/features/production/single_redirect_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      expect("config/environments/production.rb").to match_contents(
+      	middleware_canonical_host
+      )
+    end
+  end
+
+  context "destroy" do
+    it "removes Rack::CanonicalHost from the production middleware" do
+    	with_app { destroy("suspenders:production:single_redirect") }
+    	middleware_canonical_host = %r{config.middleware.use Rack::CanonicalHost, ENV.fetch\("APPLICATION_HOST"\)}
```
[the remark this answers is not in the document]
Comment (mike-burns) [reply spec/features/production/single_redirect_spec.rb]: Yup, it sure is too long. This is the length of the line.

File: lib/suspenders/generators/production/single_redirect.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require_relative "../base"
+
+module Suspenders
+  module Production
+    class SingleRedirect < Generators::Base
+      def add_rack_canonical_host
+        inject_into_file(
+        "config/environments/production.rb",
+        %{\n  config.middleware.use Rack::CanonicalHost, ENV.fetch("APPLICATION_HOST")},
```
[the remark this answers is not in the document]
Comment (mike-burns) [reply lib/suspenders/generators/production/single_redirect.rb]: I'm OK with this.

File: lib/suspenders/generators/production/single_redirect.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require_relative "../base"
+
+module Suspenders
+  module Production
+    class SingleRedirect < Generators::Base
+      def add_rack_canonical_host
+        inject_into_file(
+          "config/environments/production.rb",
+          %{\n  config.middleware.use Rack::CanonicalHost, ENV.fetch("APPLICATION_HOST")},
```
Comment (SViccari) [inline lib/suspenders/generators/production/single_redirect.rb]: Did you intend for `ENV.fetch("APPLICATION_HOST")` to be evaluated? If so, I believe you need to update this to read: `#{ENV.fetch("APPLICATION_HOST")}`

Comment (SViccari) [reply lib/suspenders/generators/production/single_redirect.rb]: ah, disregard :) I see that the intent is to add `ENV.fetch("APPLICATION_HOST")` and not the retrieved value. Makes sense 👍

**Comment by composerinteralia** [comment]
CI now failing because of https://github.com/thoughtbot/suspenders/pull/976. If you rebase I expect it will pass.

**Pull request merged**